api/src/main/java/org/openmrs/messagesource/impl/CachedMessageSource.java

```diff
@@ -54,10 +54,10 @@ public void addPresentation(PresentationMessage message) {
 	 * @see org.openmrs.messagesource.MutableMessageSource#getPresentations()
 	 */
 	public Collection<PresentationMessage> getPresentations() {
-		Collection<PresentationMessage> allMessages = new Vector<PresentationMessage>();
+		Collection<PresentationMessage> allMessages = new ArrayList<PresentationMessage>();
 		
-		for (Locale locale : localizedMap.keySet()) {
-			PresentationMessageMap codeMessageMap = localizedMap.get(locale);
+		for (Entry<String, Locale> entry : localizedMap.entrySet()) {
```

Comment:
The 'for' should be simply replaced with allMessages.addAll(localizedMap.values());